### PR TITLE
Update to storj.io/uplink

### PR DESCRIPTION
**Pull request by stefanbenten**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,16 +1,17 @@
 module storj.io/linksharing
 
-go 1.14
+go 1.13
```
Comment (ifraixedes) [inline go.mod:3]: :thinking: shouldn't we keep it in 1.14?

Comment (stefanbenten) [reply go.mod:3]: Definitely not. Neither of the packages requires it and thus we should not force everyone to update their go version.

File: linksharing/handler.go [hunk 2]
```diff
@@ -101,20 +94,15 @@ func (handler *Handler) serveHTTP(w http.ResponseWriter, r *http.Request) (err e
 		}
 	}()
 
-	b, err := p.OpenBucket(ctx, bucket, access.EncryptionAccess)
-	if err != nil {
-		handler.handleUplinkErr(w, "open bucket", err)
+	b, err := p.StatBucket(ctx, bucket)
```
Comment (jtolio) [inline linksharing/handler.go:97]: why do we need this StatBucket call?

File: linksharing/handler.go [hunk 3]
```diff
@@ -101,20 +94,15 @@ func (handler *Handler) serveHTTP(w http.ResponseWriter, r *http.Request) (err e
...
 		return err
 	}
-	defer func() {
-		if err := b.Close(); err != nil {
-			handler.log.With(zap.Error(err)).Warn("unable to close bucket")
-		}
-	}()
 
-	o, err := b.OpenObject(ctx, unencPath)
+	o, err := p.DownloadObject(ctx, bucket, key, nil)
```
Comment (jtolio) [inline linksharing/handler.go:103]: this should just be StatObject, considering the httpranger.ServeContent call doesn't use o.Read at all and only uses o.Info

File: linksharing/handler.go (hunk 2 above)
Comment (stefanbenten) [reply linksharing/handler.go]: We need the stat bucket call to return an error if the bucket does not exist. Else we would just return object not exist.

File: linksharing/handler.go (hunk 3 above)
Comment (stefanbenten) [reply linksharing/handler.go]: fixed!

File: linksharing/handler.go (hunk 2 above)
Comment (jtolio) [reply linksharing/handler.go]: yeah returning object not exist is totally acceptable, considering the StatBucket call adds a couple hundred ms to the overall operation. getting rid of this for performance for the normal case is worth it

**Pull request merged**